### DISCUSSION TITLE
Location key is not `default` on the initial location

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [remix-run/remix#8103] [Q&A]
Hi, 

According to [docs](https://github.com/remix-run/history/blob/main/docs/api-reference.md#locationkey), the location `key` should be `default` on the initial loading.

I tried accessing that by opening a new window, but it always gives me a unique key. Maybe I'm misunderstanding something, but I'm trying to check whether I can go back in history or should I do a normal navigation:

```tsx
const { key } = useLocation()
const navigate = useNavigate()

const onClick = () => {
  navigate(key === 'default' ? -1 : "/")
}
```

Any ideas? Thanks!

**Comment by gustavopch** [comment]
I don't know about `key`, but you can also check `window.history.state.idx === 0`.

BTW, Safari [seems to have implemented](https://twitter.com/dannymoerkerke/status/1725286725482611151) the Navigation API in their Tech Preview, so we'll soon probably be able to just use [`navigation.canGoBack()`](https://developer.mozilla.org/en-US/docs/Web/API/Navigation/canGoBack). Let's hope Firefox doesn't delay too much.

**Comment by 1xiscool** [comment]
you are correct it should be default on the initial loading hence `key === 'default' ` means there is no history you are saying if there is no history go back with the following line
`const onClick = () => {
  navigate(key === 'default' ? -1 : "/")
}`
instead try
`const onClick = () => {
  navigate(key !== 'default' ? -1 : "/")
}`

**Comment by muchisx** [comment]
@piotrkulpinski Experiencing the same, did you find a solution?
